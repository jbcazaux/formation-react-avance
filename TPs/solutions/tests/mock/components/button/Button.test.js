import { render, fireEvent, screen } from '@testing-library/react'
import Button from 'components/button/Button'

describe('NiceDice', () => {
  const mock = jest.fn()

  it('renders DiceRoll', () => {
    const { container } = render(<Button onClick={mock} />)

    expect(container).toMatchSnapshot()
  })

  it('calls onclick', () => {
    render(<Button onClick={mock} />)
    const button = screen.getByTestId('button')

    fireEvent.click(button)

    expect(mock).toHaveBeenCalled()
  })
})
