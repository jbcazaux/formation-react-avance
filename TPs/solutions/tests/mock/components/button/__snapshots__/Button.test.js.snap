// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NiceDice renders DiceRoll 1`] = `
<div>
  <button
    data-testid="button"
  />
</div>
`;
